storagenode/piecestore/endpoint.go

```diff
@@ -419,6 +434,21 @@ func (endpoint *Endpoint) Upload(stream pb.DRPCPiecestore_UploadStream) (err err
 	}
 }
 
+// isCongested identifies state of congestion. If the total number of
+// connections is above 80% of the MaxConcurrentRequests, then it is defined
+// as congestion.
+func (endpoint *Endpoint) isCongested() bool {
+	// congestionThreshold defines the proportion of MaxConcurrentRequests
+	// that must be reached before the storage node will begin dropping slow
+	// upload connections to preserve its own bandwidth.
+	const congestionThreshold = 0.8
```

Comment:
I wonder if that should not be a configuration variable as well?

Reply:
done